Test/CoreSDK.Test/Shared/Channel/TelemetryBufferTest.cs

```diff
@@ -26,23 +26,28 @@ public void DefaultValueIsAppropriateForProductionEnvironmentAndUnitTests()
         {
             var buffer = new TelemetryBuffer();
             Assert.Equal(500, buffer.Capacity);
+            Assert.Equal(1000000, buffer.MaximumUnsentBacklogSize);
         }
 
         [TestMethod]
         public void CanBeSetByChannelToTunePerformance()
         {
             var buffer = new TelemetryBuffer();
             buffer.Capacity = 42;
+            buffer.MaximumUnsentBacklogSize = 9999;
             Assert.Equal(42, buffer.Capacity);
+            Assert.Equal(9999, buffer.MaximumUnsentBacklogSize);
         }
 
         [TestMethod]
         public void WhenNewValueIsLessThanOneSetToDefault()
         {
             var buffer = new TelemetryBuffer();
             buffer.Capacity = 0;
+            buffer.MaximumUnsentBacklogSize = 0;
```

Comment:
Setting it to 0 but getting 1 million is not an intuitive behavior.
Other places in code it uses 0 to disable a feature, not to default it to a high value.